examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+    }
+
+
+"""
+## Build and train model
+
+The model is built and trained in this section.
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
+minimal generalization error.
+"""
+
+
+def train(train_data, train_labels, val_data, val_labels, model):
+
+    # Train model.
+    # Prepare callbacks.
+    # Path where to save best weights.
+
+    # Take the file name from the wrapper.
+    file_path = "/tmp/best_model_weights.h5"
+
+    # Initialize model checkpoint callback.
+    model_checkpoint = keras.callbacks.ModelCheckpoint(
+        file_path,
+        monitor="val_loss",
+        verbose=0,
+        mode="min",
+        save_best_only=True,
+        save_weights_only=True,
+    )
+
+    # Initialze early stopping callback.
+    # The model performance is monitored across the unseen data and stops training
+    # when the generalization error cease to decrease.
+    early_stopping = keras.callbacks.EarlyStopping(
+        monitor="val_loss", patience=10, mode="min"
+    )
+
+    # Compile model.
+    model.compile(
+        optimizer="adam", loss="sparse_categorical_crossentropy", metrics=["accuracy"],
+    )
+
+    # Fit model.
+    model.fit(
+        train_data,
+        train_labels,
+        validation_data=(val_data, val_labels),
+        epochs=20,
+        class_weight=compute_class_weights(train_labels),
+        batch_size=1,
+        callbacks=[early_stopping, model_checkpoint],
+        verbose=0,
+    )
+
+    # Load best weights.
+    model.load_weights(file_path)
+
+    return model
+
+
+# Building model(s).
+instance_shape = train_data[0][0].shape
+models = [create_model(instance_shape) for _ in range(ENSEMBLE_AVG_COUNT)]
+
+# Show single model architecture.
+print(models[0].summary())
+
+# Training model(s).
+trained_models = [
+    train(train_data, train_labels, val_data, val_labels, model)
+    for model in tqdm(models)
+]
+
+"""
+## Model evaluation
+
+The models are in their deterministic state now and ready for evaluation.
+With each model we also create an associated intermediate model to get the
+weights from the attention layer.
+
+Based on the number of models (`ENSEMBLE_AVG_COUNT`), the models predict the results
+and then averaged out (equal contribution per model).
+"""
+
+
+def predict(data, labels, trained_models):
+
+    # Collect info per model.
+    models_predictions = []
+    models_attention_weights = []
+    models_losses = []
+    models_accuracies = []
+
+    for model in trained_models:
+
+        # Predict output classes on data.
+        predictions = model.predict(data)
+        models_predictions.append(predictions)
+
+        # Create intermediate model to get MIL attention layer weights.
+        intermediate_model = keras.Model(model.input, model.get_layer("alpha").output)
+
+        # Predict MIL attention layer weights.
+        intermediate_predictions = intermediate_model.predict(data)
+
+        # Reshape list of arrays.
```

Comment:
Avoid comments like this where you are just restating what is obvious from the code below. I can tell we are reshaping from the line, and this comment doesn't communicate anything about why or how.

Reply:
Updated.